homeassistant/components/renault/sensor.py

```diff
@@ -0,0 +1,277 @@
+)
+from .renault_hub import RenaultHub
+from .renault_vehicle import RenaultVehicleProxy
+
+ATTR_BATTERY_AVAILABLE_ENERGY = "battery_available_energy"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Renault entities from config entry."""
+    proxy: RenaultHub = hass.data[DOMAIN][config_entry.unique_id]
+    entities = await get_entities(proxy)
+    async_add_entities(entities)
+
+
+async def get_entities(proxy: RenaultHub) -> list[RenaultDataEntity]:
+    """Create Renault entities for all vehicles."""
+    entities = []
+    for vehicle in proxy.vehicles.values():
+        entities.extend(await get_vehicle_entities(vehicle))
+    return entities
+
+
+async def get_vehicle_entities(vehicle: RenaultVehicleProxy) -> list[RenaultDataEntity]:
+    """Create Renault entities for single vehicle."""
+    entities: list[RenaultDataEntity] = []
+    if "cockpit" in vehicle.coordinators:
+        entities.append(RenaultMileageSensor(vehicle, "Mileage"))
+        if vehicle.details.uses_fuel():
+            entities.append(RenaultFuelAutonomySensor(vehicle, "Fuel Autonomy"))
+            entities.append(RenaultFuelQuantitySensor(vehicle, "Fuel Quantity"))
+    if "hvac_status" in vehicle.coordinators:
+        entities.append(RenaultOutsideTemperatureSensor(vehicle, "Outside Temperature"))
+    if "battery" in vehicle.coordinators:
+        entities.append(RenaultBatteryLevelSensor(vehicle, "Battery Level"))
+        entities.append(RenaultChargeStateSensor(vehicle, "Charge State"))
+        entities.append(
+            RenaultChargingRemainingTimeSensor(vehicle, "Charging Remaining Time")
+        )
+        entities.append(RenaultChargingPowerSensor(vehicle, "Charging Power"))
+        entities.append(RenaultPlugStateSensor(vehicle, "Plug State"))
+        entities.append(RenaultBatteryAutonomySensor(vehicle, "Battery Autonomy"))
+        entities.append(RenaultBatteryTemperatureSensor(vehicle, "Battery Temperature"))
+    if "charge_mode" in vehicle.coordinators:
+        entities.append(RenaultChargeModeSensor(vehicle, "Charge Mode"))
+    return entities
+
+
+class RenaultBatteryAutonomySensor(RenaultBatteryDataEntity, SensorEntity):
+    """Battery autonomy sensor."""
+
+    _attr_icon = "mdi:ev-station"
+    _attr_unit_of_measurement = LENGTH_KILOMETERS
+
+    @property
+    def state(self) -> int | None:
+        """Return the state of this entity."""
+        return self.data.batteryAutonomy if self.data else None
+
+
+class RenaultBatteryLevelSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Battery Level sensor."""
+
+    _attr_device_class = DEVICE_CLASS_BATTERY
+    _attr_unit_of_measurement = PERCENTAGE
+
+    @property
+    def state(self) -> int | None:
+        """Return the state of this entity."""
+        return self.data.batteryLevel if self.data else None
+
+    @property
+    def icon(self) -> str:
```

Comment:
This isn't needed when setting device class. It's already handled by the frontend. Please remove it.